### PR TITLE
Add examples for using Integration logging Resources and Data Sources

### PR DESCRIPTION
**Pull request by gabfelp**
## Description of the change
In this PR we add examples in the docs for `cyral_integration_logging` Data Source and Resource

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] Jira issue referenced in commit message and/or PR title

### Testing

The files are looking good via preview tool.

### CODE REVIEW

File: examples/resources/cyral_integration_logging/resource.tf [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    region = "us-east-1"
+    group  = "everyone"
+    stream = "abc"
+  }
+}
+
+resource "cyral_integration_logging" "fluent_bit_integration" {
+  name = "my-fluentbit-integration"
+  fluent_bit {
+    # Configuring a s3 bucket
```
Comment (ccampo133) [inline examples/resources/cyral_integration_logging/resource.tf:13]:
```suggestion
    # Configures a custom Fluent Bit output config to write logs to an S3 bucket
```

File: examples/resources/cyral_integration_logging/resource.tf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    config = <<-EOF
+    [OUTPUT]
+      Name s3
+      Match *
+      Region us-east-2
+      Bucket user-us-east-2
```
Comment (ccampo133) [inline examples/resources/cyral_integration_logging/resource.tf:19]:
```suggestion
      Bucket example-bucket
```

File: examples/resources/cyral_integration_logging/resource.tf [hunk 3]
```diff
@@ -0,0 +1,36 @@
+resource "cyral_integration_logging" "cloud_watch_integration" {
+  name = "my-cloudwatch-integration"
+  cloud_watch {
+    region = "us-east-1"
+    group  = "everyone"
+    stream = "abc"
```
Comment (ccampo133) [inline examples/resources/cyral_integration_logging/resource.tf:6]:
```suggestion
    group  = "example-loggroup"
    stream = "example-logstream"
```

File: docs/resources/integration_logging.md
Comment (ccampo133) [inline docs/resources/integration_logging.md:1]: I believe this file is generated? If not, see my comments on `examples/resources/cyral_integration_logging/resource.tf` and apply them here as well please.

**Review by ccampo133 — APPROVED** [review]

File: docs/resources/integration_logging.md
Comment (gabfelp) [reply docs/resources/integration_logging.md:1]: Yes, it is! Thanks!.

File: examples/resources/cyral_integration_logging/resource.tf (hunk 3 above)
Comment (gabfelp) [reply examples/resources/cyral_integration_logging/resource.tf]: Addr [feef9d5](https://github.com/cyralinc/terraform-provider-cyral/pull/400/commits/feef9d511d3c85cbf8f789150bcf28d848ae3e0b)

File: examples/resources/cyral_integration_logging/resource.tf (hunk 1 above)
Comment (gabfelp) [reply examples/resources/cyral_integration_logging/resource.tf]: Addr [feef9d5](https://github.com/cyralinc/terraform-provider-cyral/pull/400/commits/feef9d511d3c85cbf8f789150bcf28d848ae3e0b)

File: docs/data-sources/integration_logging.md [hunk 4]
```diff
@@ -1,15 +1,20 @@
...
 
 Retrieve and filter logging integrations.
 
+## Example Usage
+
+```terraform
+data "cyral_integration_logging" "some_data_source_name" {
+  # Filter you can apply
+  type = "CLOUDWATCH"
+}
```
Comment (wcmjunior) [inline docs/data-sources/integration_logging.md:11]:
```suggestion
# Return all existing CloudWatch integrations
data "cyral_integration_logging" "some_data_source_name" {
  type = "CLOUDWATCH"
}
```

File: docs/data-sources/integration_logging.md [hunk 5]
```diff
@@ -1,15 +1,20 @@
...
+
+data "cyral_integration_logging" "another_data_source_name" {
+  # type defaults to `ANY` if not set.
```
Comment (wcmjunior) [inline docs/data-sources/integration_logging.md:14]:
```suggestion
# Returns all existing integrations. Attribute `type` defaults to `ANY` if not set.
data "cyral_integration_logging" "another_data_source_name" {
```

File: docs/resources/integration_logging.md [hunk 6]
```diff
@@ -1,15 +1,47 @@
...
+
+```terraform
+resource "cyral_integration_logging" "cloud_watch_integration" {
+  name = "my-cloudwatch-integration"
+  cloud_watch {
+    region = "us-east-1"
+    group  = "example-loggroup"
+    stream = "example-logstream"
+  }
+}
```
[the remark this answers is not in the document]
Comment (gabfelp) [reply docs/resources/integration_logging.md:15]: The last one (fluentbit) worked just fine:

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # cyral_integration_logging.s3 will be created
  + resource "cyral_integration_logging" "s3" {
      + id                 = (known after apply)
      + name               = "my-s3"
      + receive_audit_logs = false

      + fluent_bit {
          + config = <<-EOT
                [OUTPUT]
                  Name s3
                  Match *
                  Region us-east-2
                  Bucket example-bucket
                  Total_file_size 1M
            EOT
        }
    }

  # cyral_sidecar.sidecar_fluent_bit will be created
  + resource "cyral_sidecar" "sidecar_fluent_bit" {
      + bypass_mode        = "failover"
      + deployment_method  = "terraform"
      + id                 = (known after apply)
      + log_integration_id = (known after apply)
      + name               = "my-sidecar"
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

cyral_integration_logging.s3: Creating...
cyral_integration_logging.s3: Creation complete after 3s [id=XXXX]
cyral_sidecar.sidecar_fluent_bit: Creating...
cyral_sidecar.sidecar_fluent_bit: Creation complete after 2s [id=XXXX]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
```

But the cloudwatch have a cycle dependency 🤔 . What we did before for this integration type ?

[the remark this answers is not in the document]
Comment (gabfelp) [reply docs/resources/integration_logging.md]: Addr here: [e01561e](https://github.com/cyralinc/terraform-provider-cyral/pull/400/commits/e01561e450d31f1f5c6b403f008187380325f6f8) created a custom name for the cloud_watch group 👍🏻

File: docs/data-sources/integration_logging.md (hunk 5 above)
Comment (gabfelp) [reply docs/data-sources/integration_logging.md]: Addr [e01561e](https://github.com/cyralinc/terraform-provider-cyral/pull/400/commits/e01561e450d31f1f5c6b403f008187380325f6f8)

File: docs/data-sources/integration_logging.md (hunk 4 above)
Comment (gabfelp) [reply docs/data-sources/integration_logging.md]: Addr [e01561e](https://github.com/cyralinc/terraform-provider-cyral/pull/400/commits/e01561e450d31f1f5c6b403f008187380325f6f8)

**Comment by gabfelp** [comment]
I added the `receive_audit_logs` conflict with `fluent_bit` in this commit: [0e39036](https://github.com/cyralinc/terraform-provider-cyral/pull/400/commits/0e3903669c7133bd8f5faf27184cf366f6280a6b)

FYI @wcmjunior @ccampo133

File: docs/resources/integration_logging.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (wcmjunior) [reply docs/resources/integration_logging.md]: > But the cloudwatch have a cycle dependency 🤔 . What we did before for this integration type ?
OMG. Indeed. I did not pay attention to that. 🤦 
My goal was to exemplify using the common name we use in our TF AWS EC2 module.

Use the following instead:

```
# Configures `my-sidecar` to push logs to CloudWatch to a log stream named `cyral-sidecar` in
# a log group created by our AWS EC2 module for deploying a sidecar.
locals {
    cloudwatch_log_group_name = "my-log-group"
}

resource "cyral_sidecar" "sidecar" {
    name = "my-sidecar"
    deployment_method = "terraform"
    log_integration_id = cyral_integration_logging. cloudwatch.id
}

resource "cyral_integration_logging" "cloudwatch" {
  name = "my-cloudwatch"
  cloud_watch {
    region = "us-east-1"
    group  = local.cloudwatch_log_group_name
    stream = "cyral-sidecar"
  }
}

module "cyral_sidecar" {
    source  = "cyralinc/sidecar-ec2/aws"  
    version = "~> 4.0"

    sidecar_version = ""
    sidecar_id      = cyral_sidecar.sidecar.id

    cloudwatch_log_group_name = local.cloudwatch_log_group_name
    ...
}
```

File: cyral/model_integration_logging.go [hunk 7]
```diff
@@ -94,7 +94,7 @@ func getIntegrationLogsSchema() map[string]*schema.Schema {
 			Default:       false,
 			ConflictsWith: []string{FluentbitKey},
 		},
-		"cloud_watch": {
+		"cloudwatch": {
```
Comment (wcmjunior) [inline cyral/model_integration_logging.go:97]: Use the const `CloudWatchKey` here instead.

**Review by wcmjunior — APPROVED** [review]

File: cyral/model_integration_logging.go (hunk 7 above)
Comment (gabfelp) [reply cyral/model_integration_logging.go]: Addr here: [c8f4bb8](https://github.com/cyralinc/terraform-provider-cyral/pull/400/commits/c8f4bb8547bfd8b49604637c8dafa9b2f3474096). 

Thanks for splitting the examples!

File: docs/resources/integration_logging.md [hunk 8]
```diff
@@ -1,15 +1,98 @@
...
+
+  ...
+}
+```
+
+### Custom Integration
+
+Custom log integrations can be defined using the `fluent_bit` block and providing output plugins.
+More information can be found in our [public docs](https://cyral.com/docs/v4.7/integrations/siem/overview)
+and in the official [Fluent Bit documentation](https://docs.fluentbit.io/manual/pipeline/outputs).
```
Comment (ccampo133) [inline docs/resources/integration_logging.md:57]:
```suggestion
### Fluent Bit

Advanced users can use the "Fluent Bit" logging integration to customize how the sidecar sends logs to their log management destination. Using the `fluent_bit`, users can provide a customized Fluent Bit configuration (in ["classic mode" INI format](https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/classic-mode)), using any combination of filter and output plugins. More information can be found in our in the official Fluent Bit [documentation](https://docs.fluentbit.io/manual/concepts/data-pipeline) (see the "Filter" and "Output" sections).
```

File: templates/resources/integration_logging.md.tmpl
Comment (ccampo133) [inline templates/resources/integration_logging.md.tmpl:1]: This should be updated to match the wording in `docs/resources/integration_logging.md` @gabfelp

File: examples/resources/cyral_integration_logging/custom.tf
Comment (ccampo133) [inline examples/resources/cyral_integration_logging/custom.tf:1]: I would rename this to `fluent-bit.tf`

File: examples/resources/cyral_integration_logging/built-in.tf
Comment (ccampo133) [inline examples/resources/cyral_integration_logging/built-in.tf:1]: I would rename this to `logging.tf`

**Pull request merged**